torch/nn/modules/distance.py

```diff
@@ -1,6 +1,25 @@
+import torch
+from .module import Module
+
+
+class PairwiseDistance(Module):
+    r"""Computes the distance between two vectors using the p-norm.
+    """
+
+    def __init__(self, p=2):
+        super(PairwiseDistance, self).__init__()
+        assert p % 1 == 0
+        self.norm = p
+
+    def forward(self, x1, x2):
+        assert x1.size() == x2.size()
+        eps = 1e-4 / x1.size(1)
+        diff = torch.abs(x1 - x2)
+        out = torch.pow(diff, self.norm).sum(dim=1)
+        return torch.pow(out + eps, 1. / self.norm)
```

Comment:
Not sure about it in here, but maybe this is not a good expression for the `1`-norm when `x1 == x2`.
In torch, the gradients in such case are randomly assigned to `-1` or `1`, [as can be seen here](https://github.com/torch/nn/blob/master/PairwiseDistance.lua#L37).
In here, the gradients are always assigned to `1` when `x1 == x2` (because `eps` is positive).

Not sure how it can affect things though.